### PR TITLE
Use package spans in partial evaluation

### PR DESCRIPTION
**Pull request by cesarzc**

### CODE REVIEW

**Review by swernli — APPROVED** [review]

File: compiler/qsc_partial_eval/tests/calls.rs [hunk 1]
```diff
@@ -1285,7 +1285,7 @@ fn call_to_unresolved_callee_producing_dynamic_value_fails() {
 
     assert_error(
         &error,
-        &expect!["UnexpectedDynamicValue(Span { lo: 298, hi: 305 })"],
+        &expect!["UnexpectedDynamicValue(PackageSpan { package: PackageId(2), span: Span { lo: 298, hi: 305 } })"],
```
Comment (swernli) [inline compiler/qsc_partial_eval/tests/calls.rs:1288]: Could you add a test case there that calls into something from stdlib that will fail with `PackageId(1)` to confirm that behavior?

File: compiler/qsc_partial_eval/src/lib.rs [hunk 2]
```diff
@@ -74,29 +76,49 @@ pub enum Error {
...
     #[error("an unexpected error occurred related to: {0}")]
     #[diagnostic(code("Qsc.PartialEval.Unexpected"))]
     #[diagnostic(help(
         "this is probably a bug, please consider reporting this as an issue to the development team"
     ))]
-    Unexpected(String, #[label] Span),
+    Unexpected(String, #[label] PackageSpan),
 
-    #[error("failed to evaluate: {0} not yet implemented")]
+    #[error("failed to evaluate: {0} is not supported yet")]
```
Comment (minestarks) [inline compiler/qsc_partial_eval/src/lib.rs:99]: Nit: I don't like the word "yet" in error messages. Never promise features in an error message :)

Comment (minestarks) [reply compiler/qsc_partial_eval/src/lib.rs:99]:
```suggestion
    #[error("failed to evaluate: {0} is not supported")]
```

File: compiler/qsc_partial_eval/src/lib.rs
Comment (minestarks) [inline compiler/qsc_partial_eval/src/lib.rs:1]: I may have missed them, but are there any tests for partial eval errors returned from the standard library?

**Review by minestarks — APPROVED** [review]

File: compiler/qsc_partial_eval/src/lib.rs
Comment (cesarzc) [reply compiler/qsc_partial_eval/src/lib.rs:1]: I think there is not, I'll add one

File: compiler/qsc_partial_eval/tests/calls.rs (hunk 1 above)
Comment (cesarzc) [reply compiler/qsc_partial_eval/tests/calls.rs:1288]: I'll add one :)

Comment (cesarzc) [reply compiler/qsc_partial_eval/tests/calls.rs:1288]: Added test.

File: compiler/qsc_partial_eval/src/lib.rs
Comment (cesarzc) [reply compiler/qsc_partial_eval/src/lib.rs:1]: Added test.

**Pull request merged**